components/brave_ads/browser/notification_helper_mac.mm

```diff
@@ -64,7 +69,52 @@
 
 ///////////////////////////////////////////////////////////////////////////////
 
-bool NotificationHelperMac::IsNotificationsEnabled() const {
+bool NotificationHelperMac::IsAuthorized() const {
+#if !defined(OFFICIAL_BUILD)
+  LOG(WARNING) << "Unable to detect the status of native notifications on non"
+      " official builds as the app is not code signed";
+  return true;
+#else
+  // TODO(https://openradar.appspot.com/27768556): We must mock this function
+  // using NotificationHelperMock as a workaround to UNUserNotificationCenter
+  // throwing an exception during tests
+
+  if (@available(macOS 10.14, *)) {
+    __block bool is_authorized = false;
+
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+    UNUserNotificationCenter *notificationCenter =
+        [UNUserNotificationCenter currentNotificationCenter];
+
+    [notificationCenter requestAuthorizationWithOptions:UNAuthorizationOptionAlert
+        completionHandler:^(BOOL granted, NSError * _Nullable error) {
+      if (granted) {
+        LOG(INFO) << "User granted authorization to show notifications";
+      } else {
+        LOG(WARNING) << "User denied authorization to show notifications";
+      }
+
+      is_authorized = granted;
+
+      dispatch_semaphore_signal(semaphore);
+    }];
+
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
This (and the same usage below) is fine for now, but we should move away from this ASAP, as _if_ Apple change the callbacks from UNUserNotificationCenter to execute async on main thread, this wait will block main thread and thus this process forever.

Fine with this as long as its a temporary solution.